### PR TITLE
Remove AttributeKeyValue, make map insert/upsert safe by copying the given value

### PR DESCRIPTION
**Pull request by bogdandrutu**
* Fix a bug in the setters for AttributeValue, the setter needs to overwrite the previous value with the default value.

### CODE REVIEW

File: internal/data/common.go [hunk 1]
```diff
@@ -108,118 +110,70 @@ func (a AttributeValue) BoolVal() bool {
 	return a.orig.BoolValue
 }
 
-func (a AttributeValue) SetString(v string) {
+func (a AttributeValue) SetStringVal(v string) {
+	a.copyValues(emptyAttributeKeyValue)
 	a.orig.Type = otlpcommon.AttributeKeyValue_STRING
 	a.orig.StringValue = v
```
Comment (tigrannajaryan) [inline internal/data/common.go:116]: Alternatively it is possible to do:
```go
*a.orig = otlpcommon.AttributeKeyValue{Type:..., StringValue...}
```
This seems shorter and no need for copyValue call or for emptyAttributeKeyValue var.

File: internal/data/common.go [hunk 2]
```diff
@@ -287,30 +241,147 @@ func (am AttributeMap) Delete(key string) bool {
...
+// the raw value to avoid an extra allocation.
+func (am AttributeMap) Insert(k string, v AttributeValue) {
+	if _, existing := am.Get(k); !existing {
+		*am.orig = append(*am.orig, newAttributeKeyValue(k, v))
+	}
+}
+
+// Insert adds the string Value to the map when the key does not exist.
+// No action is applied to the map where the key already exists.
+func (am AttributeMap) InsertString(k string, v string) {
```
Comment (tigrannajaryan) [inline internal/data/common.go:257]: What is the benefit of using InsertString vs Insert?

Comment (bogdandrutu) [reply internal/data/common.go:257]: you don't need to do an extra allocation of the AttributeValue that will be thrown away anyway.

File: internal/data/common.go (hunk 1 above)
Comment (tigrannajaryan) [reply internal/data/common.go:116]: Strange. Let's keep your code, it is fine. This may be a frequent operation so let's not loose performance.

**Pull request merged**